system-addon/lib/TopSitesFeed.jsm

```diff
@@ -104,6 +104,7 @@ this.TopSitesFeed = class TopSitesFeed {
             NewTabUtils.activityStreamProvider._addFavicons([copy]));
           copy.__sharedCache.updateLink("favicon", copy.favicon);
           copy.__sharedCache.updateLink("faviconSize", copy.faviconSize);
+          copy.__sharedCache.updateLink("faviconRef", copy.faviconRef);
```

Comment:
nit: Ref before Size here too.. Although a refactoring to avoid the missed entry would be nice. E.g.,
```js
const CACHED_LINK_PROPS_TO_MIGRATE = ["screenshot"];
const PINNED_FAVICON_PROPS_TO_MIGRATE = ["favicon", "faviconRef", "faviconSize"];

frecentCache = new LinksCache(…, CACHED_LINK_PROPS_TO_MIGRATE, …
pinnedCache = new LinksCache(…, [...CACHED_LINK_PROPS_TO_MIGRATE, ...PINNED_FAVICON_PROPS_TO_MIGRATE], …

for (const prop of PINNED_FAVICON_PROPS_TO_MIGRATE) {
  copy.__sharedCache.updateLink(prop, copy[prop]);
}
```